components/services/tor/tor_launcher_impl.cc

```diff
@@ -29,6 +29,8 @@ void TorLauncherImpl::Cleanup() {
     return;
   in_shutdown_ = true;
 
+  // Delete watch folder every time that Tor is terminated
```

Comment:
what would happen if the browser crashed and so weren't able to cleanup?

Reply:
It is actually the utility process doing the cleaning. If for some reasons utility process crash and failed to cleanup, the situation would be like what we have right now(some users might get stuck in disconnected state), and next time Tor window open and close will do the job.